src/ui/src/common/extension_command.ts

```diff
@@ -38,56 +58,54 @@ export declare interface AdapterConvertCommand extends ExtensionCommand {
   settings: Record<string, any>;
   // Whether to delete the model file at `modelPath` after conversion is done.
   deleteAfterConversion: boolean;
+  perf_trace?: string;
+  perf_data?: NodeDataProviderData;
 }
 
 /** Adapter's "convert" command response. */
-export declare interface AdapterConvertResponse extends ExtensionResponse {
-  graphs?: Graph[];
-  graphCollections?: GraphCollection[];
-}
+export type AdapterConvertResponse = ExtensionResponse;
 
 /** Adapter's "override" command. */
 export declare interface AdapterOverrideCommand extends ExtensionCommand {
   cmdId: 'override';
-  modelPath: string;
   settings: {
     graphs: Graph[];
     changes: ChangesPerNode;
   };
-  deleteAfterConversion: boolean;
 }
 
 /** Adapter's "override" command response. */
-export declare interface AdapterOverrideResponse extends ExtensionResponse {
+export type AdapterOverrideResponse = ExtensionResponse<[{
   success: boolean;
-  graphs?: Graph[];
-}
+}], never>;
 
 /** Adapter's "execute" command. */
 export declare interface AdapterExecuteCommand extends ExtensionCommand {
   cmdId: 'execute';
-  modelPath: string;
-  settings: Record<string, any>;
-  deleteAfterConversion: boolean;
 }
 
+/** Adapter's "execute" results inside the response. */
+export interface AdapterExecuteResults {}
+
 /** Adapter's "execute" command response. */
-export declare interface AdapterExecuteResponse extends ExtensionResponse, ExecutionCommand {
-}
+export type AdapterExecuteResponse = ExtensionResponse<[], never>;
 
 /** Adapter's "status check" command. */
 export declare interface AdapterStatusCheckCommand extends ExtensionCommand {
   cmdId: 'status_check';
-  modelPath: string;
-  settings: Record<string, any>;
-  deleteAfterConversion: boolean;
 }
 
-/** Adapter's "status check" command response. */
-export declare interface AdapterStatusCheckResponse extends ExtensionResponse {
+/** Adapter's "status check" results inside the response. */
+export interface AdapterStatusCheckResults {
   isDone: boolean;
   progress: number;
   total?: number;
   timeElapsed?: number;
   currentStatus?: string;
+  error?: string;
+  stdout?: string;
+  log_file?: string;
```

Comment:
Here are the properties described:

<dl>
  <dt>isDone<dt>
  <dd>If the execution process have finished, we use this property to stop pooling for updates.</dd>

  <dt>progress</dt>
  <dd>A number ranging from 0 to total, meaning how far into the execution process we are. It is used to show the loading indicator on the UI.</dd>

  <dt>total</dt>
  <dd>The maximum value the progress can go to, it defaults to 100. It is used to show the loading indicator on the UI.</dd>

  <dt>timeElapsed</dt>
  <dd>It is intended to represent how much time have passed since the start of the execution process, the idea is to show users how much time it takes to execute a model.</dd>

  <dt>currentStatus</dt>
  <dd>A string representing what part of the process is happening, for example: "loading dependencies", "applying overrides", "compiling model", etc.</dd>

  <dt>error</dt>
  <dd>The idea was that if some error have happened, then that property would have a description of the error. But I think errors are covered elsewhere.</dd>

  <dt>stdout</dt>
  <dd>The output from the server to <code>stdout</code> from the previous status update call to this  call. The idea here is to output more information to the log window.</dd>
  <dt>log_file</dt>
  <dd>This was carried over from a previous iteration, I think it was meant to point to where on the server the logs for this execution are saved, but we haven't used it, so I don't know.</dd>
</dl>

Note that all of the properties with `?` are optional. And we can change or remove any property if it doesn't make sense.